### PR TITLE
TGP-1308 addressed review comments given by performance analyst

### PR DESCRIPTION
**Pull request by AmarkumarJagtap**

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/service/RemoveRecordService.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ class RemoveRecordService @Inject() (
     hc: HeaderCarrier
   ): Future[Either[EisHttpErrorResponse, Int]] = {
     val correlationId     = uuidService.uuid
-    val requestedDateTime = dateTimeService.timestamp.asStringSeconds
+    val requestedDateTime = dateTimeService.timestamp.asStringMilliSeconds
```
Comment (taylors-version) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/service/RemoveRecordService.scala:41]: Is this changing what goes into the header to EIS?

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/service/UpdateRecordService.scala [hunk 2]
```diff
@@ -45,21 +45,21 @@ class UpdateRecordService @Inject() (
   )(implicit hc: HeaderCarrier): Future[Either[EisHttpErrorResponse, CreateOrUpdateRecordResponse]] = {
     val correlationId     = uuidService.uuid
     val payload           = UpdateRecordPayload(eori, recordId, request)
-    val requestedDateTime = dateTimeService.timestamp.asStringSeconds
+    val requestedDateTime = dateTimeService.timestamp.asStringMilliSeconds
```
Comment (taylors-version) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/service/UpdateRecordService.scala:48]: Same question

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditUpdateRecordResponse.scala [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.audit.response
+
+import play.api.libs.json._
+
+case class AuditUpdateRecordResponse(
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditUpdateRecordResponse.scala:21]: this is the same as the AuditCreateRecordResponse. You could reuse the same one for both create and Update audit

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditCreateRecordResponse.scala [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.audit.response
+
+import play.api.libs.json._
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditCreateRecordResponse.scala:19]: this is the same as the AuditUpdateRecordResponse. You could reuse the same one for both create and Update audit

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditUpdateRecordRequest.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.audit.request
+
+import play.api.libs.json.{Json, OFormat}
+
+import java.time.Instant
+
+case class AuditUpdateRecordRequest(
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditUpdateRecordRequest.scala:23]: this is the same as the AuditCreateRecordRequest. You could reuse the same one for both create and Update audit

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditCreateRecordRequest.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.audit
+
+import play.api.libs.json._
+
+import java.time.Instant
+
+case class AuditCreateRecordRequest(
```
Comment (mauro-fanton) [inline app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditCreateRecordRequest.scala:23]: this is the same as the AuditUpdateRecordRequest. You could reuse the same one for both create and Update audit

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/service/RemoveRecordService.scala (hunk 1 above)
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/service/RemoveRecordService.scala:41]: nope, it is only used for audit request/response datetime

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/service/UpdateRecordService.scala (hunk 2 above)
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/service/UpdateRecordService.scala:48]: same as above

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditUpdateRecordResponse.scala (hunk 3 above)
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditUpdateRecordResponse.scala:21]: nope, AuditCreateRecordResponse has recordId and AuditUpdateRecordResponse doesn't

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditCreateRecordResponse.scala (hunk 4 above)
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/response/AuditCreateRecordResponse.scala:19]: nope, AuditCreateRecordResponse has recordId and AuditUpdateRecordResponse doesn't

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditUpdateRecordRequest.scala (hunk 5 above)
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditUpdateRecordRequest.scala:23]: nope, AuditUpdateRecordRequest has recordId and AuditCreateRecordRequest doesn't

File: app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditCreateRecordRequest.scala (hunk 6 above)
Comment (AmarkumarJagtap) [reply app/uk/gov/hmrc/tradergoodsprofilesrouter/models/audit/request/AuditCreateRecordRequest.scala:23]: nope, AuditUpdateRecordRequest has recordId and AuditCreateRecordRequest doesn't

**Pull request merged**